sycl/include/sycl/ext/oneapi/experimental/address_cast.hpp

```diff
@@ -22,8 +22,19 @@ template <access::address_space Space, access::decorated DecorateAddress,
 multi_ptr<ElementType, Space, DecorateAddress>
 static_address_cast(ElementType *Ptr) {
 #ifdef __SYCL_DEVICE_ONLY__
-  auto CastPtr = sycl::detail::spirv::GenericCastToPtr<Space>(Ptr);
-  return multi_ptr<ElementType, Space, DecorateAddress>(CastPtr);
+  // TODO: Remove this restriction.
+  static_assert(std::is_same_v<ElementType, remove_decoration_t<ElementType>>,
+                "The extension expect undecorated raw pointers only!");
```

Comment:
Out of curiosity: is this something that would be difficult to fix?  Why not just use `remove_decoration_t` to create an undecorated raw pointer here, and then use the undecorated pointer for the rest of the function?

Reply:
I plan to do that in the next PR(s). It has to come with the doc change as well.